src/damavand/cloud/aws/resources/glue_component.py

```
@@ -118,28 +375,77 @@ def iceberg_database(self) -> aws.glue.CatalogDatabase:
             location_uri=f"s3://{self.iceberg_bucket.bucket}/",
         )
 
+    def _create_glue_trigger(
+        self, job: GlueJobDefinition
+    ) -> Optional[aws.glue.Trigger]:
+        """Creates a Glue Trigger for the job if scheduled.
+
+        Args:
+            job (GlueJobDefinition): The job definition.
+
+        Returns:
+            Optional[aws.glue.Trigger]: The Glue trigger if scheduled, else None.
+        """
+        return (
+            aws.glue.Trigger(
+                f"{job.name}-glue-trigger",
+                type="SCHEDULED",
+                schedule=job.schedule,
+                actions=[
+                    aws.glue.TriggerActionArgs(
+                        job_name=self._get_job_name(job),
+                    )
+                ],
+                start_on_creation=True,
+            )
+            if job.schedule
+            else None
+        )
+
     @property
     @cache
-    def jobs(self) -> list[aws.glue.Job]:
-        """Return all the Glue jobs for the application."""
+    def kafka_checkpoint_bucket(self) -> Optional[aws.s3.BucketV2]:
```

You should raise an attribute error if this being called while there is no streaming job instead make it optional.

---

Done